### DISCUSSION TITLE
blog/cloud/2024-07-01-CI-CD-pipeline/

### DISCUSSION THREAD

**Discussion by giscus** [mingdodev/mingdodev.github.io#18] [Announcements]
# blog/cloud/2024-07-01-CI-CD-pipeline/

한번 배포에 성공하더라도 변경 사항이 생길 때마다 개발자가 여러 가지 작업을 직접 해줘야 하니 몹시 번거로웠다. 코드 저장소의 내용을 변경하는 것만으로 알아서 배포가 이루어질 수 있도록 자동화 도구를 사용해보고자 CI/CD에 관해 찾아보게 되었다. 마침 ACC 초반 세션 중 제대로 정리가 안 되어있는 부분을 발견해서 복습할 겸 추가 공부를 해보려고 한다.

https://mingdodev.github.io/blog/cloud/2024-07-01-CI-CD-pipeline/

**Comment by hwanheejung** [comment]
github actions 많이 사용하던데 사용법도 궁금하네용

**Reply by mingdodev** [reply]
자동화 기술이 없으니 굉장히 불편하길래 조만간 직접 써보려구요! github actions를 사용하게 된다면 블로그에 자세하게 써볼게요~~

**Comment by mikio999** [comment]
애자일하군요.. 그렇다면 CI/CD의 단점도 존재할까요?

**Reply by mingdodev** [reply]
알아보시는군요..(?)

결국 CI/CD가 반복 작업을 줄여주는 것은 맞지만 이를 위한 초기 세팅이 복잡하다는 점이 단점 같아요. 사람이 맘 편히 손을 놓으려면 언제든 오류가 발생하지 않을 정확한 설계가 필요하므로, 고려해야 할 요소가 단순 배포 시점보다 훨씬 많아지는 느낌입니다. 하지만 역시 가장 큰 단점은 비용 문제라고 생각합니다. 편리한 기술을 사용하는 만큼 대가를 지불해야만 하는.. 사용해본 후 좀 더 현실적인 문제들에 대해서도 후기 남겨볼게요!
